Create book

ShastkivRuslan opened the pull request:
Add Book entity, repository, service, and CommandLineRunner for initial setup

- Created Book entity with fields: id, title, author, isbn, price, description, coverImage
- Implemented BookRepository interface and BookRepositoryImpl class
- Added BookService interface and BookServiceImpl class with methods to save and find all books
- Configured @SpringBootApplication class with CommandLineRunner to initialize the app
- Set up required properties
- switched to lambda.

ShastkivRuslan committed:
Created User model and add required fields without annotations;

ShastkivRuslan committed:
Added all of required annotations for User model;
Added lombok dependency;

ShastkivRuslan committed:
Add Book entity, repository, service, and CommandLineRunner for initial setup

- Created Book entity with fields: id, title, author, isbn, price, description, coverImage
- Implemented BookRepository interface and BookRepositoryImpl class
- Added BookService interface and BookServiceImpl class with methods to save and find all books
- Configured @SpringBootApplication class with CommandLineRunner to initialize the app
- Set up required properties

ShastkivRuslan committed:
Add Book entity, repository, service, and CommandLineRunner for initial setup

- Created Book entity with fields: id, title, author, isbn, price, description, coverImage
- Implemented BookRepository interface and BookRepositoryImpl class
- Added BookService interface and BookServiceImpl class with methods to save and find all books
- Configured @SpringBootApplication class with CommandLineRunner to initialize the app
- Set up required properties
- switched to lambda.

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/BookstoreApplication.java:
```diff
@@ -1,13 +1,33 @@
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return args -> {
+            Book modernJava = new Book();
```
```suggestion
            Book book = new Book();
```

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/exception/DataProcessingException.java:
```diff
@@ -0,0 +1,6 @@
+package ruslan.shastkiv.bookstore.exception;
+
+public class DataProcessingException extends RuntimeException {
+    public DataProcessingException(String s, Exception e) {
```
rename s

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/model/Book.java:
```diff
@@ -0,0 +1,33 @@
+
+@Entity
+@Table(name = "users")
+@Data
+@NoArgsConstructor
```
```suggestion
@Getter
@Setter
```

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/model/Book.java:
```diff
@@ -0,0 +1,33 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NonNull
```
Let's be consistent
```suggestion
    @Column(nullable = false)
```
fix for other fields

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,54 @@
+import ruslan.shastkiv.bookstore.model.Book;
+import ruslan.shastkiv.bookstore.repository.BookRepository;
+
+@Repository
```
```suggestion
@Repository
@RequiredArgsConstructor
```

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,54 @@
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+
```
```suggestion
```

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,54 @@
+
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
```suggestion
```

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/service/impl/BookServiceImpl.java:
```diff
@@ -0,0 +1,29 @@
+import ruslan.shastkiv.bookstore.repository.BookRepository;
+import ruslan.shastkiv.bookstore.service.BookService;
+
+@Service
```
```suggestion
@Service
@RequiredArgsConstructor
```

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/service/impl/BookServiceImpl.java:
```diff
@@ -0,0 +1,29 @@
+
+@Service
+public class BookServiceImpl implements BookService {
+
```
```suggestion
```

Elena-Bruyako commented on src/main/java/ruslan/shastkiv/bookstore/service/impl/BookServiceImpl.java:
```diff
@@ -0,0 +1,29 @@
+
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
```suggestion
```

Elena-Bruyako commented on pom.xml:
```diff
@@ -36,12 +37,31 @@
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.34</version>
+            <scope>provided</scope>
```
```suggestion
```

Elena-Bruyako reviewed (changes requested):
see comments

ShastkivRuslan committed:
- renamed "modernJava" variable to "book" in CommandLineRunner;
-changed exception signature;
-removed "@NoArgsConstructor", "@NonNull" and "@Data" annotations in Book class;
-added "@Getter" and "@Setter" annotations in Book class;
-changed "@NonNull" annotation to "@Column(nullable = false)";
-removed constructors with "@Autowired" annotations in BookRepositoryImpl and BookServiceImpl class;
-added "@RequiredArgsConstructor" to BookRepositoryImpl and BookServiceImpl class;
-deleted some info from pom.xml;

ShastkivRuslan committed:
-fixed 2 checkstyle errors;)

Elena-Bruyako reviewed (approved):
well done
